satpy/readers/ahi_hsd.py

```diff
@@ -506,7 +536,7 @@ def read_band(self, key, info):
         logger.debug("Reading time " + str(datetime.now() - tic))
 
         # Calibrate
-        res = self.calibrate(res, key.calibration)
+        res = self.calibrate(res, key.name, key.calibration)
```

Comment:
Instead of passing the name as an argument, could you make it an instance attribute? The hsd format has only on band per file. Moreover, please use a more readable name, like 'bandname'.

Reply:
You mean remove `key.name` and make the file handler know which band will be read in `__init__`. My main argument against that is that it prevents more flexible code in the future including possibly sharing HSD reader code or sharing AHI reader code. Is it possible that HSD formats could be joined together in future or someone could make their own HSD-like format that might have more than one band? Or that we read data quality (or similar) fields from the HSD flag in the same way that the band data is read?

Reply:
I see your point, but I would argue that this is premature optimisation. Also, seeing how the industry works, I highly doubt the hsd format will be reused: in 99% of the cases, they come up with a new format. Especially considering this is a binary format. But indeed, maybe there are already other hsd derived formats I'm unaware of. Lastly, this is a pattern we use in other readers, so it won't be an isolated case.

Reply:
Sorry, could you explain what you mean "an instance attribute"? I don't know what that is or how it'd work. You mean add it in `__init`` as `self.name = 'blah'` or something?

Reply:
> Lastly, this is a pattern we use in other readers, so it won't be an isolated case.

Hm I was going to argue the same point about the existing implementation. Having "helper" functions that `get_dataset` calls with the current key/id information. Honestly, it isn't that big of a deal and I see now that `key.name` didn't exist before so it makes sense that it shouldn't be added here. I was going to argue that the code is already implemented this way so why not leave it, but see that it is a new argument/implementation and see what you're saying now.

Reply:
@djhoese happy we agree in the end :)

@simon_p yes, `self.bandname = fileinfo[...] `

Reply:
Done!